types/index.d.ts

```diff
@@ -876,3 +879,8 @@ declare namespace jest {
   // eslint-disable-next-line @typescript-eslint/no-empty-interface
   interface InverseAsymmetricMatchers extends Expect {}
 }
+
+declare namespace Vi {
+  // eslint-disable-next-line @typescript-eslint/no-empty-interface
+  interface AsymmetricMatchersContaining extends CustomMatchers {}
+}
```

Comment:
Took the idea from

https://github.com/vitest-dev/vitest/blob/199776767a10bcd9f241e36a5cdd578c7d7c0dc2/test/core/test/jest-expect.test.ts#L9-L17